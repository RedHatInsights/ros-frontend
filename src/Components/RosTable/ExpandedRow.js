import React from 'react';
import propTypes from 'prop-types';
import {
    DescriptionList,
    DescriptionListTerm,
    DescriptionListGroup,
    DescriptionListDescription
} from '@patternfly/react-core';
import { NO_DATA_VALUE } from '../../constants';

export const ExpandedRow = ({ inventoryId, cloudProvider, instanceType, idlingTime }) =>  {

    const cloudProviderValue = cloudProvider === null ? NO_DATA_VALUE : cloudProvider;
    const instanceTypeValue = instanceType === null ? NO_DATA_VALUE : instanceType;
    const idlingTimeValue = idlingTime === null ? NO_DATA_VALUE : `${idlingTime}%`;

    const inlineFieldsArray = [
        { label: 'Provider', value: cloudProviderValue },
        { label: 'Instance type', value: instanceTypeValue },
        { label: 'Idling time', value: idlingTimeValue }
    ];

    return (
        <DescriptionList className='expanded-row' isCompact isHorizontal>
            <DescriptionListGroup id={inventoryId}>
                {inlineFieldsArray.map((field, index) => (
                    <React.Fragment key={index}>
                        <DescriptionListTerm>{ field.label }</DescriptionListTerm>
                        <DescriptionListDescription>{ field.value }</DescriptionListDescription>
                    </React.Fragment>
                ))}
            </DescriptionListGroup>
        </DescriptionList>
    );
};

ExpandedRow.propTypes = {
    inventoryId: propTypes.string,
    cloudProvider: propTypes.string,
    instanceType: propTypes.string,
    idlingTime: propTypes.string
};
