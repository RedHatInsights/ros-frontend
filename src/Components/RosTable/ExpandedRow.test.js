import React from 'react';
import { cleanup } from '@testing-library/react';
import { ExpandedRow } from './ExpandedRow';
import renderer from 'react-test-renderer';
import { DescriptionListGroup } from '@patternfly/react-core';

describe('ExpandedRow component', () => {
    afterEach(cleanup);

    it ('matches snapshot', ()=>{
        const propValues = {
            inventoryId: 'fd480a4c-952b-4675-ba02-3370bd08a5ee',
            cloudProvider: 'aws',
            instanceType: 't2.micro',
            idlingTime: '1.03'
        };

        const component = renderer.create(<ExpandedRow { ...propValues }/>);
        expect(component.toJSON()).toMatchSnapshot();
        const expandedRowInstance = component.root;
        expect(expandedRowInstance.findByType(DescriptionListGroup).props.id).toBe(propValues.inventoryId);
    });
});
