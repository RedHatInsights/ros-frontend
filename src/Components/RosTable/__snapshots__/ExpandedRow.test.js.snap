// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpandedRow component matches snapshot 1`] = `
<dl
  className="pf-c-description-list pf-m-horizontal pf-m-compact expanded-row"
>
  <div
    className="pf-c-description-list__group"
    id="fd480a4c-952b-4675-ba02-3370bd08a5ee"
  >
    <dt
      className="pf-c-description-list__term"
    >
      <span
        className="pf-c-description-list__text"
      >
        Provider
      </span>
    </dt>
    <dd
      className="pf-c-description-list__description"
    >
      <div
        className="pf-c-description-list__text"
      >
        aws
      </div>
    </dd>
    <dt
      className="pf-c-description-list__term"
    >
      <span
        className="pf-c-description-list__text"
      >
        Instance type
      </span>
    </dt>
    <dd
      className="pf-c-description-list__description"
    >
      <div
        className="pf-c-description-list__text"
      >
        t2.micro
      </div>
    </dd>
    <dt
      className="pf-c-description-list__term"
    >
      <span
        className="pf-c-description-list__text"
      >
        Idling time
      </span>
    </dt>
    <dd
      className="pf-c-description-list__description"
    >
      <div
        className="pf-c-description-list__text"
      >
        1.03%
      </div>
    </dd>
  </div>
</dl>
`;
